src/components/sidebar/channel-members/ChannelMembers.js

```diff
@@ -7,27 +7,72 @@ import User from './User'
 export default class ChannelMembers extends PureComponent {
   static propTypes = {
     users: PropTypes.array.isRequired,
+    totalMembers: PropTypes.number,
     channel: PropTypes.object.isRequired,
     currUser: PropTypes.object.isRequired,
+    sidebarRef: PropTypes.oneOfType([PropTypes.func, PropTypes.object]),
     onLoad: PropTypes.func,
     onOpen: PropTypes.func,
     onKick: PropTypes.func,
   }
 
   static defaultProps = {
+    totalMembers: 0,
+    sidebarRef: undefined,
     onLoad: noop,
     onOpen: noop,
     onKick: noop,
-    users: [],
+  }
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      limit: 50,
+      shift: 50,
+      // A height of each member item is 42. 10 of them it's 420.
+      bottomOffset: 420,
+    }
   }
 
   componentDidMount() {
-    this.props.onLoad()
+    this.props.onLoad(true)
+    this.props.sidebarRef.addEventListener('scroll', this.onScroll, true)
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const { shift } = this.state
+    if (shift > nextProps.users.length) {
+      this.setState({
+        shift: 50,
+      })
+    }
   }
 
   componentDidUpdate(nextProps) {
     if (nextProps.channel.id !== this.props.channel.id) {
-      this.props.onLoad()
+      this.props.onLoad(true)
+    }
+  }
+
+  componentWillUnmount() {
+    this.props.sidebarRef.removeEventListener('scroll', this.onScroll, true)
+  }
+
+  onScroll = e => {
+    const { offsetHeight, scrollTop, scrollHeight } = e.target
+    const { users, totalMembers } = this.props
+    const { shift, limit, bottomOffset } = this.state
+    if (
+      shift > users.length ||
```

Comment:
In which case can shift be larger than the loaded user length?

Reply:
When `shift` already incremented, but data didn't load